### PR TITLE
New random generator

### PR DESCRIPTION
**Pull request by jonano614**
New random generator based on implementation from bitcoin

### CODE REVIEW

File: client/crypt.c [hunk 1]
```diff
@@ -29,15 +29,9 @@ extern unsigned int xOPENSSL_ia32cap_P[4];
 extern int xOPENSSL_ia32_cpuid(unsigned int *);
 
 // initialization of the encryption system
-int xdag_crypt_init(int withrandom)
```
Comment (xrdavies) [inline client/crypt.c]: Not sure what is `withrandom` used for.

File: client/utils/random.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	QueryPerformanceCounter((LARGE_INTEGER*)&counter);
+#else
+	struct timeval t;
+	gettimeofday(&t, NULL);
+	counter = (int64_t)(t.tv_sec * 1000000 + t.tv_usec);
+#endif
+	return counter;
+}
+
+void RandAddSeed()
```
Comment (sgaragagghu) [inline client/utils/random.c]: If i understood correctly, in unix systems the seed is took from /dev/random and windows with RAND_screen();
After this, to add more entropy to the seed, it add the time. Is it right ?

File: client/crypt.c (hunk 1 above)
Comment (jonano614) [reply client/crypt.c]: if `withrandom` is not zero, OpenSSL random generator was seeded by some random array generated based on dnet key. 
Function out_balances was calling xdag_crypt_init with `withrandom` = 0

File: client/utils/random.c (hunk 2 above)
Comment (jonano614) [reply client/utils/random.c]: I'm not sure about /dev/random, but yes, time stamp is used for seeding the OpenSSL random generator. It is also possible to use some screen data for seeding on Windows OS.

**Pull request merged**